src/main/scala/org/deepdive/ddlog/DeepDiveLogPrettyPrinter.scala

```diff
@@ -48,16 +48,61 @@ object DeepDiveLogPrettyPrinter extends DeepDiveLogHandler {
        |""".stripMargin
   }
 
+  // print an expression
+  def printExpr(e: Expr) : String = {
+    e match {
+      case VarExpr(name) => name
+      case ConstExpr(value) => {
+        if (value.startsWith("'")) s""" "${value.stripPrefix("'").stripSuffix("'")}" """
+        else value
+      }
+      case FuncExpr(function, args, agg) => {
+        val resolvedArgs = args map (x => printExpr(x))
+        s"${function}(${resolvedArgs.mkString(", ")})"
+      }
+      case BinaryOpExpr(lhs, op, rhs) => s"${printExpr(lhs)} ${op} ${printExpr(rhs)}"
+    }
+  }
+
+  // print a condition
+  def printCond(cond: Cond) : String = {
+    cond match {
+      case ComparisonCond(lhs, op, rhs) => s"${printExpr(lhs)} ${op} ${printExpr(rhs)}"
+      case NegationCond(c) => s"![${printCond(c)}]"
+      case BinaryOpCond(lhs, op, rhs) => {
+        op match {
+          case LogicOperator.AND => s"[${printCond(lhs)}], [${printCond(rhs)}]" 
+          case LogicOperator.OR  => s"[${printCond(lhs)}]; [${printCond(rhs)}]"
```

Comment:
I think it's equivalent, but adding brackets for lhs and rhs is more natural. An binary operator means taking whatever in the left and right, then connect them using the operator, so lhs and rhs are evaluated as a whole. Precisely, `AND(P, Q)` should be evaluated as `[P] AND [Q]`, not `[P AND Q]` which is equivalent to `P AND Q`. If needed, the brackets are added by its upper-level function call.